src/benchq/resource_estimation/graph/transformers.py

```diff
@@ -84,3 +86,33 @@ def _transformer(program: QuantumProgram) -> GraphPartition:
         return GraphPartition(new_program, [graph])
 
     return _transformer
+
+
+def remove_isolated_nodes(graph_partition: GraphPartition) -> GraphPartition:
+    """Sometimes our circuits can generate a lot of extra nodes because of how
+    RESET adds nodes to the graph. This transformer removes these nodes from the
+    graph to prevent them from influencing the costing. There are 3 known sources
+    of isolated nodes:
+        1. Unneeded resets at the beginning of the circuit.
+        2. Decomposing rotations into gates sometimes gives bare nodes if a
+             T gate is placed after a reset. (most concerning)
```

Comment:
Oh, these are concerning because it seems like there is a failure in the algorithm. This means that some of the T gates are more or less unused despite being included in the resource count.